lib/fog/vsphere/models/compute/server.rb

```diff
@@ -230,6 +230,7 @@ def add_interface(attrs)
 
         def update_interface(attrs)
           wait_for { !ready? } if interface_ready? attrs
+          attrs = attrs.merge(datacenter: datacenter) unless attrs.key? :datacenter
```

Comment:
You can use `attrs[:datacenter] = datacenter` here. That avoids the call to `merge`.

Reply:
changed